examples/04.MQTT/mqtt.cc

```diff
@@ -17,12 +17,12 @@ using CHERI::Capability;
 using Debug            = ConditionalDebug<true, "MQTT example">;
 constexpr bool UseIPv6 = CHERIOT_RTOS_OPTION_IPv6;
 
-// The client ID is used by the broker to identify the client. It should be
-// unique per client. Since this is hard-coded, this means we cannot launch
-// multiple concurrent instances of this example on a same test server.
-// TODO it would be nice to generate automatically.
-constexpr const char  *clientID       = "anoncheri42";
-constexpr const size_t clientIDLength = sizeof(clientID);
+constexpr const int clientIDLength = 23;
+char                clientID[clientIDLength];
+
+constexpr const char clientIDPrefix[]     = "cheriotMQTT";
+constexpr const int  clientIDPrefixLength = sizeof(clientIDPrefix) - 1 /* \0 */;
+static_assert(clientIDPrefixLength < clientIDLength);
```

Comment:
It's better style to make this correct by construction, rather than to add asserts that check that you got it right.  For example:

```c++
/// Prefix for MQTT client identifier
constexpr const char clientIDPrefix[]     = "cheriotMQTT";
/// Length of client identifier 
constexpr const int  clientIDPrefixLength = sizeof(clientIDPrefix) - 1;
/// Length of the client ID including the suffix
constexpr const int clientIDLength = clientIDPrefixLength + 12;
/// Space for the random client ID.
char                clientID[clientIDLength];
```

This can be further simplified as:

```c++
/// Prefix for MQTT client identifier
constexpr std::string_view clientIDPrefix{"cheriotMQTT"};
/// Length of the client ID including the suffix
constexpr const int clientIDLength = clientIDPrefix.size() + 12;
/// Space for the random client ID.
std::array<char, clientIDLength> clientID;
```

EDIT: It looks as if the 23 number came from the maximum client ID length permitted by MQTT.  In that case, that should be a symbolic constant and then we possibly do need the static assert, but then the code looks like this:

```c++
/// Maximum permitted MQTT client identifier length (from the MQTT specification)
constexpr size_t MQTTMaximumClientLength = 23;
/// Prefix for MQTT client identifier
constexpr std::string_view ClientIDPrefix{"cheriotMQTT"};
/// Space for the random client ID.
std::array<char, MQTTMaximumClientLength> clientID;
static_assert(ClientIDPrefix.size() < clientID.size);
```